### PR TITLE
Security 1.4.0.2 changes

### PR DESCRIPTION
**Pull request by sujithvm**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vrozov** [comment]
Is this a cherry-pick from master? Please add
```
(cherry picked from commit <master commit id>)
```
to the commit message.

File: pom.xml [hunk 1]
```diff
@@ -98,7 +98,7 @@
         <url>https://github.com/opendistro-for-elasticsearch/security</url>
         <connection>scm:git:git@github.com:opendistro-for-elasticsearch/security.git</connection>
         <developerConnection>scm:git:git@github.com:opendistro-for-elasticsearch/security.git</developerConnection>
-        <tag>1.4.0.1</tag>
+        <tag>1.4.0.2</tag>
```
Comment (vrozov) [inline pom.xml]: Consider `${project.version}`. Is `tag` used by the build?

File: opendistro-elasticsearch-security.release-notes [hunk 2]
```diff
@@ -1,4 +1,9 @@
-## 2020-03-05, Version 1.4.0.1 (Current)
+## 2020-03-31, Version 1.4.0.2 (Current)
+- Fixed bug which caused user to lose roles on account password update
```
[the remark this answers is not in the document]
Comment (sujithvm) [reply opendistro-elasticsearch-security.release-notes:3]: I don't know why it is not .md 
Maybe fix it separately?

File: pom.xml (hunk 1 above)
Comment (sujithvm) [reply pom.xml:101]: Looks like scm:tag is not correctly used.  Looking up online it seems to be using HEAD as common value. 
Maybe when the security repos were separate, this was used to push to local mvn repository and used for building there. 

Maybe investigate this and put the correct value separately as `${project.version}` may not be correct as well

File: opendistro-elasticsearch-security.release-notes (hunk 2 above)
[the remark this answers is not in the document]
Comment (vrozov) [reply opendistro-elasticsearch-security.release-notes:3]: I was curious to understand the reason, no need to change it on the branch.

**Pull request merged**